extensions/wikia/ImageReview/modules/PromoteImage/PromoteImageReviewTask.php

```diff
@@ -1,429 +1,429 @@
+			$sourceWikiLang = WikiFactory::getVarValueByName('wgLanguageCode', $sourceWikiId);
+
+			if( !empty($images) ) {
+				$removedImages = array();
+				foreach($images as $imageName) {
+					if (PromoImage::fromPathname($imageName)->isValid()) {
+						$result = $this->removeSingleImage($corpWikiId, $imageName);
+
+						if( $result['status'] === 0 ) {
+							//almost all the time on devboxes images aren't removed because of no permissions
+							//when we run maintenance/wikia/ImageReview/PromoteImage/remove.php with sudo it works
+							$removedImages[] = $imageName;
+						}
+					}
+				}
+			}
+
+			if( !empty($removedImages) ) {
+				$memcKey = $this->helper->getMemcKey($sourceWikiId, $sourceWikiLang);
+				$updateData = $this->syncAdditionalImages($sourceWikiId, $sourceWikiLang, $removedImages);
+
+				//update in db
+				if( !empty($updateData) ) {
+					//updating city_visualization table
+					$this->model->saveVisualizationData(
+						$sourceWikiId,
+						$updateData,
+						$sourceWikiLang
+					);
+
+					//purging interstitial cache
+					$app->wg->Memc->set($memcKey, null);
+				}
+			}
+		}
+
+		//since an admin can't delete main image we don't purge visualization list cache
+		//as it happens during uploads
+		return true;
+	}
+
+	function removeSingleImage($targetWikiId, $imageName) {
+		global $IP, $wgWikiaLocalSettingsPath;
+
+		$retval = -1;
+
+		$sCommand = "SERVER_ID={$targetWikiId} php $IP/maintenance/wikia/ImageReview/PromoteImage/remove.php";
+		$sCommand .= " --imagename=" . escapeshellarg($imageName);
+		$sCommand .= " --userid=" . escapeshellarg( $this->mUser );
+		$sCommand .= " --conf {$wgWikiaLocalSettingsPath}";
+
+		$output = wfShellExec($sCommand, $retval);
+		$city_url = WikiFactory::getVarValueByName("wgServer", $targetWikiId);
+
+		if( empty($city_url) ) {
+			$this->log('Apparently the server is not available via WikiFactory');
+			return array('status' => 1);
+		}
+
+		$city_path = WikiFactory::getVarValueByName("wgScript", $targetWikiId);
+		if( $retval != 0 ) {
+			$this->log('Remove error! (' . $city_url . '). Error code returned: ' . $retval . ' Error was: ' . $output);
+		} else {
+			$this->log('Removal successful: <a href="' . $city_url . $city_path . '?title=' . wfEscapeWikiText($output) . '">' . $city_url . $city_path . '?title=' . $output . '</a>');
+		}
+
+		return array(
+			'status' => $retval,
+			'title' => $output,
+		);
+	}
+
+	protected function getImagesToUpdateInDb($sourceWikiId, $sourceWikiLang, $images) {
+		$data = array();
+
+		$wikiData = $this->model->getWikiData($sourceWikiId, $sourceWikiLang, $this->helper);
+		$currentImages = $wikiData['images'];
+
+		if( !empty($currentImages) ) {
+			foreach($currentImages as $imageName) {
+				$promoImage = PromoImage::fromPathname($imageName);
+				if( $promoImage->isAdditional() && !in_array($promoImage->getPathname(), $images) ) {
+					$data['city_images'][] = $promoImage->getPathname();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "city_images" 12 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)